test/e2e/awscodecommitsource/awscodecommitsource.go

```diff
@@ -0,0 +1,206 @@
+	"github.com/triggermesh/aws-event-sources/test/e2e/framework/awscodecommit"
+	"github.com/triggermesh/aws-event-sources/test/e2e/framework/conversion"
+	"github.com/triggermesh/aws-event-sources/test/e2e/framework/sources"
+)
+
+var _ = Describe("AWS CodeCommit source", func() {
+	var (
+		srcClient dynamic.ResourceInterface
+		ccClient  codecommitiface.CodeCommitAPI
+
+		sink     duckv1.Destination
+		repoARN  apis.ARN
+		awsCreds credentials.Value
+	)
+
+	f := framework.NewDefaultFramework("awscodecommitsource")
+
+	BeforeEach(func() {
+		srcClient = f.DynamicClient.
+			Resource(conversion.V1alpha1GVR(apisources.AWSCodeCommitSourceResource)).
+			Namespace(f.Namespace.Name)
+
+		By("reading AWS credentials", func() {
+			sess := session.Must(session.NewSession())
+			ccClient = codecommit.New(sess)
+			awsCreds = readAWSCredentials(sess)
+		})
+
+		By("creating a Git repository", func() {
+			arn, undoCreateRepo := awscodecommit.CreateRepository(ccClient, f)
+			framework.AddCleanupAction(
+				undoCreateRepo,
+			)
+			repoARN = parseARN(arn)
+		})
+
+		By("creating an event sink", func() {
+			sink = sources.CreateEventDisplaySink(f.ClientSet, f.Namespace.Name)
+		})
+	})
+
+	It("should generate events on selected actions", func() {
+		By("creating an AWSCodeCommitSource object", func() {
+			src, err := createSource(srcClient, f.Namespace.Name, "test-", sink,
+				withARN(repoARN),
+				withCredentials(awsCreds),
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			sources.WaitForResourceReady(f.DynamicClient, src)
+		})
+
+		By("creating a Git commit", func() {
+			awscodecommit.CreateCommit(ccClient, repoARN.Resource)
+		})
+
+		// TODO(antoineco): check event-display logs, metrics?
+	})
+
+	It("should reject invalid specs", func() {
+		By("creating an AWSCodeCommitSource object with an invalid ARN", func() {
+			invalidARN := repoARN
+			invalidARN.AccountID = "invalid"
+
+			_, err := createSource(srcClient, f.Namespace.Name, "test-invalid-arn", sink,
+				withARN(invalidARN),
+				withCredentials(awsCreds),
+			)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("spec.arn: Invalid value: "))
+		})
+
+		By("creating an AWSCodeCommitSource object without credentials", func() {
+			_, err := createSource(srcClient, f.Namespace.Name, "test-nocreds", sink,
+				withARN(repoARN),
+			)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring(
+				`"spec.credentials.accessKeyID" must validate one and only one schema (oneOf).`))
+		})
+
+		By("creating an AWSCodeCommitSource object with invalid event types", func() {
+			_, err := createSource(srcClient, f.Namespace.Name, "test-invalid-eventtypes", sink,
+				withARN(repoARN),
+				withEventType("invalid"),
+				withCredentials(awsCreds),
+			)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring(`spec.eventTypes: Unsupported value: "invalid"`))
+		})
```

Comment:
This is a good demonstration of how ginkgo tests are structured. `BeforeEach` can perform some initialization before every test, like creating a CodeCommit repo, or an event sink, etc.

At the end of every test the corresponding Kubernetes namespace is implicitly destroyed, regardless of the outcome of the test, so we only need to destroy external AWS resources here.